src/subgraph/portfolio/index.ts

```diff
@@ -31,7 +31,34 @@ export const getPortfolioDataByUsersAddress = async (
 
   return portfolioData;
 };
-
+export const getOpenPositionsAndDepositCollateralByAddress = async (
+  subgraphEndpoint: string,
+  usersAddress: string[],
+  collateralPrice: { id: string; price: number }[],
+) =>{
+  const subgraphResponse: PorfolioDataSubgraph = await request(subgraphEndpoint, fetchUserOpenPositionAndDepositCollateral(usersAddress));
+const openPositionData = await Promise.all(subgraphResponse.wallets?.map(async (data) => {
```

Comment:
why you are using promise and await, map function don't need to be a async function.

Reply:
so that if there is any promise in our the accountId array then it return the resolved  accountId array so we don't need to put await while reading the accountId array from the returned object

Reply:
why there will be promise in accountId array.

Reply:
yes your are right .my bad i have made map async  and was getting promise  while testing so i make it promises  all will changed it thanks

Reply:
remove await as well